src/lib/features/artifacts.js

```diff
@@ -139,8 +140,29 @@ export default class Artifacts {
       };
 
       try {
-        let digest = await uploadToS3(taskHandler.queue, taskId, runId, stream,
+        let region = taskHandler.runtime.region;
+        assert.ok(region);
+
+        let monitor = taskHandler.runtime.monitor;
+        let start = process.hrtime();
+
+        let {digest, size} = await uploadToS3(taskHandler.queue, taskId, runId, stream,
                          entryName, expiry, headers);
+
+        // save the time taken to upload the artifact
+        let elapsedTime = process.hrtime(start);
+        let uploadTime = elapsedTime[0] + elapsedTime[1] / 1e9; // in seconds
```

Comment:
let's keep speed in bytes/second and report the time in milliseconds.  With that, r+